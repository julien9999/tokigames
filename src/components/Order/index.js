import React from 'react';
import { connect } from 'react-redux';
import { setOrder } from '../../actions/orderActions';
import { SHOW_DURATION, SHOW_STARTTIME, SHOW_ENDTIME } from '../../constants/actionTypes';

function Link({ active, children, onClick }) {
  return (
    <li className="nav-item">
        <a onClick={onClick} className={`nav-link ${active ? 'active' : null}`} href="#">
          {children}
        </a>
    </li>
  )
}

const mapStateToProps = (state, ownProps) => ({
  active: ownProps.order === state.order
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  onClick: () => dispatch(setOrder(ownProps.order))
});

const FilterLink = connect(
  mapStateToProps,
  mapDispatchToProps
)(Link);

export default function () {
  return (
    <div className='container'>
        <div className='row'>
          <div className='col-12'> 
            <h5 class="card-title margin-top-twenty">Order options:</h5>
          </div>
          <div className='col-12'>       
            <ul class="nav nav-pills nav-fill">
              <FilterLink order={SHOW_DURATION}>Duration</FilterLink>
              <FilterLink order={SHOW_STARTTIME}>Start Time</FilterLink>
              <FilterLink order={SHOW_ENDTIME}>End Time</FilterLink>
            </ul>
          </div>
        </div>
    </div>
  );
}

