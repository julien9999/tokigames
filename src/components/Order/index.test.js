import React from 'react';
import renderer from 'react-test-renderer';
import Order from './index';
import { Provider } from 'react-redux';
import configure from '../../store/configureStore';

const store = configure();

describe('<Order>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
            <Order />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

