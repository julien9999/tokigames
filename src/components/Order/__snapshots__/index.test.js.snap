// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <h5
        class="card-title margin-top-twenty"
      >
        Order options:
      </h5>
    </div>
    <div
      className="col-12"
    >
      <ul
        class="nav nav-pills nav-fill"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link null"
            onClick={[Function]}
          >
            Duration
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link null"
            onClick={[Function]}
          >
            Start Time
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link null"
            onClick={[Function]}
          >
            End Time
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
